Add initial version of 3scale alert SOPs included on 3scale v2.9 (#9)
3scale/3scale-Operations: sops/alerts/apicast_apicast_latency.adoc

:toc:
:toc-placement!:

= ThreescaleApicastLatencyHigh

toc::[]

== Description

* This alert will trigger if the `apicast-staging`/`apicast-production` **p99** latency is greater than a given threshold

== Troubleshooting

> * Check if `apicast-staging`/`apicast-production` pods might not be able to deal with incoming traffic (too much cpu usage causing slowliness?)

Reviewer: ```suggestion
* Check if `apicast-staging`/`apicast-production` pods cpu usage is high as might not be able to deal with incoming traffic
```

> * Check if `apicast-staging`/`apicast-production` pods might not be able to deal with incoming traffic (too much cpu usage causing slowliness?)

Reviewer: ```suggestion
* Check if `apicast-staging`/`apicast-production` pods might not be able to deal with incoming traffic due to high CPU usage
```

- You may need to consider scaling horinzontally `apicast-staging`/`apicast-production` deployment
- You may need to consider increasing `apicast-staging`/`apicast-production` deployment resources requests/limits
* Check if upstream response is taking too much time

> 

Reviewer: ```suggestion
+ Check if the current traffic is normal based on the historical data. An unexpected increase of traffic can be legit, and will require scaling up, but it also be part of an attack, and will require countermeasures to block the malicious traffic .

```

> 

Reviewer: ```suggestion
* Check if the current traffic is normal based on the historical data. An unexpected increase of traffic can be legit, and will require scaling up, but it also be part of an attack, and will require countermeasures to block the malicious requests .

```

:toc:
:toc-placement!:

= ThreescaleApicastLatencyHigh

toc::[]

== Description

> * This alert will trigger if the `apicast-staging`/`apicast-production` **p99** latency is greater than a given threshold

Author: ```suggestion
* This alert will trigger if the `apicast-staging`/`apicast-production` **p99** latency is greater than a given threshold
* Latency metric includes the time the request spent in apicast plus the time it took the upstream to respond
```

:toc:
:toc-placement!:

= ThreescaleApicastLatencyHigh

toc::[]

== Description

* This alert will trigger if the `apicast-staging`/`apicast-production` **p99** latency is greater than a given threshold
* Latency metric includes the time the request spent in apicast plus the time it took the upstream to respond

== Troubleshooting

* Check if `apicast-staging`/`apicast-production` pods might not be able to deal with incoming traffic due to high CPU usage
- You may need to consider scaling horinzontally `apicast-staging`/`apicast-production` deployment
- You may need to consider increasing `apicast-staging`/`apicast-production` deployment resources requests/limits
* Check if upstream response is taking too much time

> * Check if the current traffic is normal based on the historical data. An unexpected increase of traffic can be legit, and will require scaling up, but it also be part of an attack, and will require countermeasures to block the malicious requests .

Author: ```suggestion
* Check if the current traffic is normal based on the historical data. An unexpected change in the traffic pattern can be legit, and will require scaling up, but it also can be due to an abnormal or malicious traffic
```
